pennylane/tape/tape.py

```diff
@@ -515,6 +527,8 @@ def _update(self):
         self._update_trainable_params()
         self._update_observables()
 
+        self._circuit = self.operations + self.measurements
```

Comment:
I think you should define an empty circuit in __init__ it should solve some failures in the test suite.